homeassistant/components/unifiprotect/media_player.py

```diff
@@ -0,0 +1,139 @@
+from homeassistant.const import STATE_IDLE, STATE_PLAYING
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .data import ProtectData
+from .entity import ProtectDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[Sequence[Entity]], None],
+) -> None:
+    """Discover cameras with speakers on a UniFi Protect NVR."""
+    data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            ProtectMediaPlayer(
+                data,
+                camera,
+            )
+            for camera in data.api.bootstrap.cameras.values()
+            if camera.feature_flags.has_speaker
+        ]
+    )
+
+
+class ProtectMediaPlayer(ProtectDeviceEntity, MediaPlayerEntity):
+    """A Ubiquiti UniFi Protect Speaker."""
+
+    def __init__(
+        self,
+        data: ProtectData,
+        camera: Camera,
+    ) -> None:
+        """Initialize an UniFi speaker."""
+
+        self.device: Camera = camera
+        self.entity_description = MediaPlayerEntityDescription(
+            key="speaker", device_class=DEVICE_CLASS_SPEAKER
+        )
+        super().__init__(data)
+
+        self._attr_name = f"{self.device.name} Speaker"
+        self._attr_supported_features = (
+            SUPPORT_PLAY_MEDIA
+            | SUPPORT_VOLUME_SET
+            | SUPPORT_VOLUME_STEP
+            | SUPPORT_STOP
+            | SUPPORT_SELECT_SOURCE
+        )
+        self._attr_media_content_type = MEDIA_TYPE_MUSIC
+
+    @callback
+    def _async_update_device_from_protect(self) -> None:
+        super()._async_update_device_from_protect()
+        self._attr_volume_level = float(self.device.speaker_settings.volume / 100)
+
+        if (
+            self.device.talkback_stream is not None
+            and self.device.talkback_stream.is_running
+        ):
+            self._attr_state = STATE_PLAYING
+        else:
+            self._attr_state = STATE_IDLE
+
+    @callback
+    async def async_set_volume_level(self, volume: float) -> None:
+        """Set volume level, range 0..1."""
+
+        volume_int = int(volume * 100)
+        await self.device.set_speaker_volume(volume_int)
+
+    async def async_media_stop(self) -> None:
+        """Send stop command."""
+
+        if (
+            self.device.talkback_stream is not None
+            and self.device.talkback_stream.is_running
+        ):
+            _LOGGER.debug("Stopping playback for %s Speaker", self.device.name)
+            await self.device.stop_audio()
+            self._async_updated_event()
+
+    async def async_play_media(
+        self, media_type: str, media_id: str, **kwargs: Any
+    ) -> None:
+        """Play a piece of media."""
+
+        if media_type != MEDIA_TYPE_MUSIC:
+            _LOGGER.warning(
```

Comment:
Nowadays we raise a ValueError if incorrect input is passed in the service call.